### PR TITLE
Use GKE specific configuration for kube-apiserver SNI cert

### PR DESCRIPTION
**Pull request by benhxy**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Allow GKE to configure VM start up differently.
More specifically, kube-apiserver SNI cert configuration is changed for GKE.

#### Which issue(s) this PR fixes:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

### CODE REVIEW

File: cluster/gce/gci/configure-kubeapiserver.sh [hunk 1]
```diff
@@ -95,6 +95,9 @@ function start-kube-apiserver {
   if [[ -n "${TLS_CIPHER_SUITES:-}" ]]; then
     params+=" --tls-cipher-suites=${TLS_CIPHER_SUITES}"
   fi
+  if [[ -e "${KUBE_HOME}/bin/gke-internal-configure-helper.sh" ]]; then
+    gke-kube-apiserver-internal-sni-param params
```
Comment (liggitt) [inline cluster/gce/gci/configure-kubeapiserver.sh:99]: modifying params by reference is unexpected and hard to follow

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -1269,9 +1269,12 @@ EOF
 function create-kubeconfig {
   local component=$1
   local token=$2
-  echo "Creating kubeconfig file for component ${component}"
-  mkdir -p "/etc/srv/kubernetes/${component}"
-  cat <<EOF >"/etc/srv/kubernetes/${component}/kubeconfig"
+  if [[ -e "${KUBE_HOME}/bin/gke-internal-configure-helper.sh" ]]; then
+    gke-internal-create-kubeconfig "${component}" "${token}"
```
Comment (liggitt) [inline cluster/gce/gci/configure-helper.sh:1273]: this is hard to follow and means a separate implementation has to match the `/etc/srv/kubernetes/${component}/kubeconfig` path constructed here

File: cluster/gce/gci/configure-kubeapiserver.sh (hunk 1 above)
Comment (benhxy) [reply cluster/gce/gci/configure-kubeapiserver.sh:99]: I saw on line 75-76. configure-etcd-params made the same choice. Any recommended alternatives?

Comment (benhxy) [reply cluster/gce/gci/configure-kubeapiserver.sh:99]: Would returning a string from the function, and concatenating params here better?

Comment (liggitt) [reply cluster/gce/gci/configure-kubeapiserver.sh:99]: that would certainly make it clearer from here that params were being modified

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (benhxy) [reply cluster/gce/gci/configure-helper.sh:1273]: How about also passing the path into the internal function? Would that be clearer?

Comment (liggitt) [reply cluster/gce/gci/configure-helper.sh:1273]: yes

Comment (benhxy) [reply cluster/gce/gci/configure-helper.sh]: Splitting this change to a different PR for dberkov.

File: cluster/gce/gci/configure-kubeapiserver.sh (hunk 1 above)
Comment (benhxy) [reply cluster/gce/gci/configure-kubeapiserver.sh]: Updated to use explicit returned string.

File: cluster/gce/gci/configure-kubeapiserver.sh [hunk 3]
```diff
@@ -95,6 +95,9 @@ function start-kube-apiserver {
   if [[ -n "${TLS_CIPHER_SUITES:-}" ]]; then
     params+=" --tls-cipher-suites=${TLS_CIPHER_SUITES}"
   fi
+  if [[ -e "${KUBE_HOME}/bin/gke-internal-configure-helper.sh" ]]; then
+    params+=" $(gke-kube-apiserver-internal-sni-param)"
```
Comment (PurelyApplied) [inline cluster/gce/gci/configure-kubeapiserver.sh:99]: I'm not that familiar with the ins and outs of these configuration scripts other than what I've had to do to debug some gnarly issues.  Does the existence of the script necessarily indicate that it has been `source`d by the time we're here?  This feels like it's begging to throw a `command not found` at us.

My initial reaction was to suggest we `source` the `*configure-helper.sh` here, since it should be safe to `source` in this `if` block even if it has already been `source`d before.  But looking around elsewhere, there are a other commands (i.e., `append-param-if-not-present`, `get-metadata-value`) that are just assumed to be on the `PATH`, so maybe that worry is moot.

**Review by PurelyApplied — COMMENTED** [review]
/ok-to-test

These sort of hook shoehorns feel gross.  `tech_debt++`.

/lgtm

File: cluster/gce/gci/configure-kubeapiserver.sh (hunk 3 above)
Comment (benhxy) [reply cluster/gce/gci/configure-kubeapiserver.sh:99]: It's sourced in https://github.com/Kubernetes/kubernetes/blob/master/cluster/gce/gci/configure.sh#L386
